evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -579,6 +579,33 @@ def test_make_component_list_from_actions():
                                                           TargetImputer(impute_strategy="most_frequent")]
 
 
+@pytest.mark.parametrize("samplers", [None, "Undersampler", "SMOTE Oversampler", "SMOTENC Oversampler", "SMOTEN Oversampler"])
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass', 'regression'])
+def test_make_pipeline_samplers(problem_type, samplers, X_y_binary, X_y_multi, X_y_regression, has_minimal_dependencies):
+    if problem_type == 'binary':
+        X, y = X_y_binary
+    elif problem_type == 'multiclass':
+        X, y = X_y_multi
+    else:
+        X, y = X_y_regression
+    estimators = get_estimators(problem_type=problem_type)
+
+    for estimator in estimators:
+        pipeline = make_pipeline(X, y, estimator, problem_type, sampler_name=samplers)
+        if has_minimal_dependencies and samplers is not None:
+            samplers = 'Undersampler'
+        # check that we do add the sampler properly
+        component_to_check = pipeline.component_graph
+        if samplers is not None and problem_type != 'regression':
+            # we add the sampler before the scaler if it exists
```

Comment:
Oh, I think I missed this detail. Why do this?

Reply:
We want the sampler to come before the scaler since the scaler turns boolean values to int values ([0, 1] -> [-1, 1]), and since literature suggests scaling data should come after sampling